daemon/daemon_test.go

```diff
@@ -163,7 +164,9 @@ func (ds *DaemonSuite) TearDownTest(c *C) {
 
 	// Release the identity allocator reference created by NewDaemon. This
 	// is done manually here as we have no Close() function daemon
-	cache.Close()
+	ds.d.identityAllocator.Close()
+
+	identitymanager.RemoveAll()
```

Comment:
What's the identity allocator vs identity manager? Both of these things sound like they manage the allocation of identities.

Reply:
The behavior of the identity manager was decoupled from the allocator because we have a case when resolving an endpoint identity that its identity is 'stale' - we allocate an identity, the identity revision becomes stale, we release that identity, and try again. We only add the identity into the identity manager if the identity is actually used for a locally running endpoint (e.g., after `SetIdentity`). Is it OK if I create a TODO and put it in the ticket to couple them together? Then as part of that ticket we can see whether it makes sense to couple them together.

Reply:
@ianvernon Yes